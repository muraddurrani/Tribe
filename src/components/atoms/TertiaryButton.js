import React from 'react'
import { StyleSheet } from 'react-native'
import { Button } from 'react-native-elements'
import theme from '../../styles/theme'

function TertiaryButton(props) {
  return (
    <Button
      disabledStyle = {props.disabledStyle}
      disabledTitleStyle = {props.disabledTitleStyle}
      disabled = {props.disabled}
      onPress = {props.onPress}
      containerStyle = {{...props.containerStyle}}
      buttonStyle = {{...styles.button, ...props.buttonStyle}}
      titleStyle = {{...styles.title, ...props.titleStyle}}
      title = {props.title}
      icon = {props.icon}
    />
  )
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: theme.colours.gray0
  },
  title: {
    color: theme.colours.gray6,
    fontSize: 14
  }
})

export default TertiaryButton
