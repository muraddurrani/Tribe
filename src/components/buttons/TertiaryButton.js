import React from 'react'
import { TouchableOpacity } from 'react-native'
import { Text } from 'react-native-elements'

function TertiaryButton({title, style, titleStyle, ...props}) {
  return (
    <TouchableOpacity {...props} style = {style}>
      <Text h4Style = {{...titleStyle}} h4>{title}</Text>
    </TouchableOpacity>
  )
}

export default TertiaryButton
