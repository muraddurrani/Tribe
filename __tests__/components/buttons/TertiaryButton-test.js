import React from 'react'
import renderer from 'react-test-renderer'
import TertiaryButton from '../../../src/components/buttons/TertiaryButton'

it('TertiaryButton renders correctly', () => {
  const tree = renderer
    .create(<TertiaryButton title = 'Tertiary Button'/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})