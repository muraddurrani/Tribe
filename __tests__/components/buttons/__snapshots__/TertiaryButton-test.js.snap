// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TertiaryButton renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#242424",
        "fontSize": 28.285714285714285,
      }
    }
  >
    Tertiary Button
  </Text>
</View>
`;
